src/Slack/Provider.php

```diff
@@ -26,6 +30,50 @@ class Provider extends AbstractProvider implements ProviderInterface
     protected $scopeSeparator = ',';
 
     /**
+     * Middleware that throws exceptions for non successful slack api calls
+     * "http_error" request option is set to true.
+     *
+     * @return callable Returns a function that accepts the next handler.
+     */
+    private function getSlackApiErrorMiddleware()
```

Comment:
For this I just followed this same structure as https://github.com/guzzle/guzzle/blob/master/src/Middleware.php#L52